src/test/documentation_examples.rs

```diff
@@ -0,0 +1,1385 @@
+    collection.drop(None).await.unwrap();
+
+    // Start Example 6
+    let docs = vec![
+        doc! {
+            "item": "journal",
+            "qty": 25,
+            "size": {
+                "h": 14,
+                "w": 21,
+                "uom": "cm"
+            },
+            "status": "A"
+        },
+        doc! {
+            "item": "notebook",
+            "qty": 50,
+            "size": {
+                "h": 8.5,
+                "w": 11,
+                "uom": "in"
+            },
+            "status": "A"
+        },
+        doc! {
+            "item": "paper",
+            "qty": 100,
+            "size": {
+                "h": 8.5,
+                "w": 11,
+                "uom": "in"
+            },
+            "status": "D"
+        },
+        doc! {
+            "item": "planner",
+            "qty": 75,
+            "size": {
+                "h": 22.85,
+                "w": 30,
+                "uom": "cm"
+            },
+            "status": "D"
+        },
+        doc! {
+            "item": "postcard",
+            "qty": 45,
+            "size": {
+                "h": 10,
+                "w": 15.25,
+                "uom": "cm"
+            },
+            "status": "A"
+        },
+    ];
+
+    collection.insert_many(docs, None).await?;
+    // End Example 6
+
+    assert_coll_count!(collection, 5);
+
+    // Start Example 7
+    let cursor = collection.find(None, None).await?;
+    // End Example 7
+
+    assert_cursor_count!(cursor, 5);
+
+    // Start Example 9
+    let cursor = collection.find(doc! { "status": "D" }, None).await?;
+    // End Example 9
+
+    assert_cursor_count!(cursor, 2);
+
+    // Start Example 10
+    let cursor = collection
+        .find(
+            doc! {
+                "status": {
+                    "$in": ["A", "D"],
+                }
+            },
+            None,
+        )
+        .await?;
+    // End Example 10
+
+    assert_cursor_count!(cursor, 5);
+
+    // Start Example 11
+    let cursor = collection
+        .find(
+            doc! {
+                "status": "A",
+                "qty": { "$lt": 30 },
+            },
+            None,
+        )
+        .await?;
+    // End Example 11
+
+    assert_cursor_count!(cursor, 1);
+
+    // Start Example 12
+    let cursor = collection
+        .find(
+            doc! {
+                "$or": [
+                    { "status": "A" },
+                    {
+                        "qty": { "$lt": 30 },
+                    }
+                ],
+            },
+            None,
+        )
+        .await?;
+    // End Example 12
+
+    assert_cursor_count!(cursor, 3);
+
+    // Start Example 13
+    let cursor = collection
+        .find(
+            doc! {
+                "status": "A",
+                "$or": [
+                    {
+                        "qty": { "$lt": 30 },
+                    },
+                    {
+                        "item": { "$regex": "^p" },
```

Comment:
Based on discussion I found on the [Python ticket](https://jira.mongodb.org/browse/PYTHON-1250) for these examples, it seems like this is explicitly supposed to be the `$regex` query operator, not the BSON type, so I left it like this.

(The fact that the operator doesn't actually require there to be an underlying BSON type is one of the several reasons that I wish BSON didn't have a native regex type...)